### ISSUE TITLE
[Bug] HGT failed on mini-batch inference

### ISSUE THREAD

**Issue by wangz10** [awslabs/graphstorm#570]
I run into the following error when running inference on a LP task using a trained HGT model:

```
Traceback (most recent call last):
  File "/develop/graphstorm/python/graphstorm/run/gsgnn_lp/lp_infer_gnn.py", line 91, in <module>
    main(args)
  File "/develop/graphstorm/python/graphstorm/run/gsgnn_lp/lp_infer_gnn.py", line 73, in main
    infer.infer(infer_data, dataloader,
  File "/develop/graphstorm/python/graphstorm/inference/lp_infer.py", line 77, in infer
    embs = do_mini_batch_inference(self._model, data, fanout=loader.fanout,
  File "/develop/graphstorm/python/graphstorm/model/gnn.py", line 893, in do_mini_batch_inference
    embeddings = dist_minibatch_inference(data.g,
  File "/develop/graphstorm/python/graphstorm/model/gnn_encoder_base.py", line 204, in dist_minibatch_inference
    out_embs[ntype][out_nodes] = output[ntype].cpu()
  File "/usr/local/lib/python3.8/dist-packages/dgl/distributed/dist_tensor.py", line 182, in __setitem__
    self.kvstore.push(name=self._name, id_tensor=idx, data_tensor=val)
  File "/usr/local/lib/python3.8/dist-packages/dgl/distributed/kvstore.py", line 1253, in push
    assert F.shape(id_tensor)[0] == F.shape(data_tensor)[0], \
AssertionError: The data must has the same row size with ID.
```

The task succeeded with `--use-mini-batch-infer false`.

**Comment by classicsong** [comment]
Is this fixed?

**Comment by zhjwy9343** [comment]
Yes, this issue was fixed by PR #569.
